components/push/android/src/main/java/org/mozilla/push/LibPushFFI.kt

```diff
@@ -39,59 +39,48 @@ internal interface LibPushFFI : Library {
     // String will work but either force us to leak them, or cause us to corrupt the heap (when we
     // free them).
 
-    /* From places
-    / ** Create a new push connection */
+    /** Create a new push connection */
     fun push_connection_new(
-            db_path: String,
-            encryption_key: String?,
+            server_host: String,
+            socket_protocol: String?,
+            bridge_type: String?,
+            registration_id: String,
+            sender_id: String?,
             out_err: RustError.ByReference
     ): RawPushConnection?
 
-    / ** Returns JSON string, which you need to free with push_destroy_string */
-    fun push_note_observation(
+    /** Returns JSON string, which you need to free with push_destroy_string */
+    fun push_get_subscription_info(
             conn: RawPushConnection,
-            json_observation_data: String,
+            channel_id: String,
             out_err: RustError.ByReference
-    )
+    ): Pointer
 
-    / ** Returns JSON string, which you need to free with push_destroy_string */
-    fun push_query_autocomplete(
+    /** Returns bool */
+    fun push_unsubscribe(
             conn: RawPushConnection,
-            search: String,
-            limit: Int,
+            channel_id: String,
             out_err: RustError.ByReference
-    ): Pointer?
+    ): Bool?
```

Comment:
For primitives we don't return `Blah?` its just `Blah`. Bool is special due to JNA not respecting the platform ABI, and you should use `Byte` instead.